README.md

```diff
@@ -1,350 +1,477 @@
+
+* sending a message
+* receiving a message
+* processing a received message
+
+To trace an outgoing message, you simply need to create a MessagingSystemInfo and call traceOutgoingMessage with that instance:
+ 
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+OutgoingMessageTracer outgoingMessageTracer = oneAgentSDK.traceOutgoingMessage(messagingSystemInfo);
+outgoingMessageTracer.start();
+try {
+	// transport the dynatrace tag along with the message: 	
+	messageToSend.setHeaderField(
+		OneAgentSDK.DYNATRACE_MESSAGE_PROPERTYNAME, outgoingMessageTracer.getDynatraceStringTag());
+	theQueue.send(messageToSend);
+	
+	// optional:  add messageid provided from messaging system
+	outgoingMessageTracer.setVendorMessageId(toSend.getMessageId());
+	// optional:  add correlationId
+	outgoingMessageTracer.setCorrelationId(toSend.correlationId);
+} catch (Exception e) {
+	outgoingMessageTracer.error(e.getMessage());
+	// rethrow or add your exception handling
+} finally {
+	outgoingMessageTracer.end();
+}
+```
+
+On the incoming side, we need to differentiate between the blocking receiving part and processing the received message. Therefore two 
+different tracers are being used: ``ReceivingMessageTracer`` and ``ProcessingMessageTracer``.
+
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+// message receiving daemon task:
+while(true) {
+	IncomingMessageReceiveTracer incomingMessageReceiveTracer = 
+		oneAgentSDK.traceIncomingMessageReceive(messagingSystemInfo);
+	incomingMessageReceiveTracer.start();
+	try {
+		// blocking call - until message is being available:
+		Message queryMessage = theQueue.receive("client queries");
+		IncomingMessageProcessTracer incomingMessageProcessTracer = oneAgentSDK
+			.traceIncomingMessageProcess(messagingSystemInfo);
+		incomingMessageProcessTracer.setDynatraceStringTag(
+			queryMessage.getHeaderField(OneAgentSDK.DYNATRACE_MESSAGE_PROPERTYNAME));
+		incomingMessageProcessTracer.setVendorMessageId(queryMessage.msgId);
+		incomingMessageProcessTracer.setCorrelationId(queryMessage.correlationId);
+		incomingMessageProcessTracer.start();
+		try {
+			// do the work ... 
+		} catch (Exception e) {
+			incomingMessageProcessTracer.error(e.getMessage());
+			Logger.logError(e);
+		} finally {
+			incomingMessageProcessTracer.end();
+		}
+	} catch (Exception e) {
+		incomingMessageReceiveTracer.error(e.getMessage());
+		// rethrow or add your exception handling
+	} finally {
+		incomingMessageReceiveTracer.end();
+	}
+}
+```
+
+In case of non-blocking receive (e. g. via eventhandler), there is no need to use ``ReceivingMessageTracer`` - just trace processing 

```

Comment:
```suggestion
In case of non-blocking receive (e. g. via eventhandler), there is no need to use `IncomingMessageReceiveTracer` - just trace processing 
```